utils/cputils.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+from typing import Optional
+import os
+import subprocess
+import pathlib
+
+
+def rename_sqlite_file(sqlite_dir_path: pathlib.Path, name: str):
+    """Rename the .sqlite file into {name}.sqlite as to differentiate between different files.
+
+    Args:
+        sqlite_dir_path (pathlib.Path): path to CellProfiler_output directory
+        name (str): new name for the SQLite file
+
+    Raises:
+        FileNotFoundError: This error will occur if you do not have a SQLite file with the hard coded file name in the specified directory.
+        This means that this function can not find the right file to rename, so it raises an error.
+    """
+    try:
+        # CellProfiler requires a name to be set in to pipeline, so regardless of plate or method, all sqlite files name are hardcoded
+        sqlite_file_path = pathlib.Path(f"{sqlite_dir_path}/NF1_data.sqlite")
+
+        new_file_name = str(sqlite_file_path).replace(
+            sqlite_file_path.name, f"{name}.sqlite"
+        )
+
+        # change the file name in the directory
+        pathlib.Path(sqlite_file_path).rename(pathlib.Path(new_file_name))
+        print(f"The file is renamed to {pathlib.Path(new_file_name).name}!")
+
+    except FileNotFoundError as e:
+        print(
+            f"The NF1_data.sqlite file is not found in directory. Either the pipeline wasn't ran properly or the file is already renamed.\n"
+            f"{e}"
+        )
+
+
+def run_cellprofiler(
+    path_to_pipeline: str,
+    path_to_output: str,
+    path_to_images: str,
+    sqlite_name: Optional[None | str] = None,
+    analysis_run: Optional[False | bool] = False,
+):
+    """Run CellProfiler on data. It can be used for either a illumination correction pipeline (default) and analysis pipeline (when
+    parameter is set to True).
+
+    Args:
+        path_to_pipeline (str): path to the CellProfiler .cppipe file with the segmentation and feature measurement modules
+        path_to_output (str): path to the output folder
+        path_to_images (str): path to the images
+        sqlite_name (str, optional): string with name for SQLite file for an analysis pipeline if you plan on running
+        multiple sets of images (e.g., per plate) so that the outputs will have different names (default is None)
+        analysis_run (bool, optional): will use functions to complete an analysis pipeline (default is False)
+    """
+    # check to make sure paths to pipeline and directory of images are correct before running the pipeline
+    if not pathlib.Path(path_to_pipeline):
+        raise FileNotFoundError(f"The file '{pathlib.Path(path_to_pipeline).name}' does not exist")
+    if not pathlib.Path(path_to_images).is_dir():
+        raise FileNotFoundError(f"Directory '{pathlib.Path(path_to_images).name}' does not exist or is not a directory")
+
+    # make logs directory
+    log_dir = pathlib.Path("./logs")
+    os.makedirs(log_dir, exist_ok=True)
```

Comment:
consider using `pathlib.Path().mkdir()`.

 ```suggestion
    log_dir.mkdir(exist_ok=True)
```

Reply:
I didn't know pathlib had that function! This has been changed.